sequence_processing_pipeline/Pipeline.py

```diff
@@ -360,20 +361,37 @@ def get_sample_ids(self):
 
         return results
 
-    def get_sample_names(self):
+    def get_sample_names(self, project_name=None):
```

Comment:
If `self.mapping_file is not None` then there is a `.unique()` applied to the sample names. If it is `None`, then `.unique()` is not applied. It looks like this was in the existing code as well as the current PR, however I'm not sure I understand why there is a difference. Could a comment be added describing specifically why there is a difference in the uniqueness requirements of the sample names?

Reply:
I think they are two different methods based on the if/else:
`_get_sample_names_from_mapping_file` vs. 
`_get_sample_names_from_sample_sheet ` ...

Reply:
Ya I know, but I don't know why uniqueness depends on whether it's a mapping file or sample sheet

Reply:
Please see the proposed addition of the `_get_sample_names_from_mapping_file` below

Reply:
> If `self.mapping_file is not None` then there is a `.unique()` applied to the sample names. If it is `None`, then `.unique()` is not applied. It looks like this was in the existing code as well as the current PR, however I'm not sure I understand why there is a difference. Could a comment be added describing specifically why there is a difference in the uniqueness requirements of the sample names?

It's a good point. I believe that post-discussions w/the wetlab that we can assume sample-names are unique in sample-files and mapping-files. We certainly expect them to be in Qiita, as far I remember. I've gone ahead and removed the unique() filter. The tests complete successfully as was expected.